test/services/UIBuilderSpec.scala

```diff
@@ -478,76 +480,57 @@ class UIBuilderSpec extends BaseSpec with ProcessJson {
       }
     }
 
-    "Convert a empty RowGroup into a table" in new TableTest {
+    "Ignore an empty RowGroup" in new TableTest {
       val p = uiBuilder.buildPage("/start", Seq(TitleCallout(headingPhrase, Seq.empty, false),
                                                 emptyRowGroup))
       p.components match {
-        case Seq(_: H1, _: Table) => succeed
+        case Seq(_: H1) => succeed
         case x => fail(s"Found $x")
       }
     }
 
-    "Convert a RowGroup of empty rows into a table" in new TableTest {
-      val p = uiBuilder.buildPage("/start", Seq(TitleCallout(headingPhrase, Seq.empty, false),
-                                                oddRowGroup))
-      p.components match {
-        case Seq(_: H1, _: Table) => succeed
-        case x => fail(s"Found $x")
-      }
-    }
-
-    "Convert a non-summarylist RowGroup into a table" in new TableTest {
+    "Convert a simple RowGroup into a NameValueSummaryList" in new TableTest {
       val p = uiBuilder.buildPage("/start", Seq(TitleCallout(headingPhrase, Seq.empty, false),
                                                 simpleRowGroup))
       p.components match {
-        case Seq(_: H1, _: Table) => succeed
-        case x => fail(s"Found $x")
-      }
-    }
-
-    "Convert a non-summarylist one row RowGroup into a table" in new TableTest {
-      val oneRow = Row(Seq(Phrase(Vector("HELLO", "HELLO")), Phrase(Vector("World", "World")), Phrase(Vector("Blah", "Blah"))), Seq())
-
-      val p = uiBuilder.buildPage("/start", Seq(TitleCallout(headingPhrase, Seq.empty, false), RowGroup(Seq(oneRow))))
-
-      p.components match {
-        case Seq(_: H1, Table(_, None, Seq())) => succeed
+        case Seq(_: H1, _: NameValueSummaryList) => succeed
         case x => fail(s"Found $x")
       }
     }
 
     "Convert a non-summarylist RowGroup into a table with a heading line" in new TableTest {
-      val p = uiBuilder.buildPage("/start", Seq(TitleCallout(headingPhrase, Seq.empty, false),
-                                                captionRowGroup))
+      println(s"IsCandidate ${tableRowGroup.isTableCandidate}")
+      val p = uiBuilder.buildPage("/start", Seq(SubSectionCallout(headingPhrase, Seq.empty, false),
+                                                tableRowGroup))
       p.components match {
-        case Seq(_: H1, Table(Text(Seq(Words("HELLO", true)), Seq(Words("HELLO", true))), None, _)) => succeed
+        case Seq(Table(Text(headingPhrase,_), tableHeading, _)) => succeed
         case x => fail(s"Found $x")
       }
     }
 
     "convert a non-summarylist RowGroup stacked to a SubSection into a table with caption and a heading" in new TableTest {
```

Comment:
I think stack is already true. The RowGroup stack value is set to the value of stack for the first row. If you make the RowGroup stack false the previous test will fail.